scripts/db_operations.sh

```
@@ -0,0 +1,215 @@
+check_db_connectivity() {
+  log_message "info" "Checking database connectivity..."
+  
+  # Define a max retry count
+  max_retries=10
+  retry_count=0
+  retry_delay=3
+  connected=false
+  
+  # Try to connect to the database
+  while [ $retry_count -lt $max_retries ] && [ "$connected" != "true" ]; do
+    if pg_isready -h $POSTGRES_HOST -p $POSTGRES_PORT -U $POSTGRES_USER; then
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)